src/Renderer/ThreeExtented/KMZLoader.js

```diff
@@ -28,60 +28,35 @@ define('Renderer/ThreeExtented/KMZLoader', ['Renderer/ThreeExtented/jszip.min',
-
-                xhr.open("GET", url, true);
-
-                xhr.responseType = "arraybuffer";
-
-                xhr.crossOrigin = '';
-
-                var scopeLoader = this.colladaLoader;
-
-                xhr.onload = function() {
-
-                    var zip = new JSZip(this.response);
-                    var collada = undefined;
-                    var coordCarto = undefined;
-                    for (var name in zip.files) {
-
-                        if (name.toLowerCase().substr(-4) === '.dae') {
-                            collada = scopeLoader.parse(zip.file(name).asText());
-                        } else if (name.toLowerCase().substr(-4) === '.kml') {
+        KMZLoader.prototype.parsCollada = function(buffer, scopeLoader) {
+            var zip = new JSZip(buffer);
+            var collada = undefined;
+            var coordCarto = undefined;
 
-                            var parser = new DOMParser();
-                            var doc = parser.parseFromString(zip.file(name).asText(), "text/xml");
+            for (var name in zip.files) {
+                if (name.toLowerCase().substr(-4) === '.dae') {
+                    collada = scopeLoader.parse(zip.file(name).asText());
+                } else if (name.toLowerCase().substr(-4) === '.kml') {
 
-                            var longitude = Number(doc.getElementsByTagName("longitude")[0].childNodes[0].nodeValue);
-                            var latitude = Number(doc.getElementsByTagName("latitude")[0].childNodes[0].nodeValue);
-                            var altitude = Number(doc.getElementsByTagName("altitude")[0].childNodes[0].nodeValue);
+                    var parser = new DOMParser();
+                    var doc = parser.parseFromString(zip.file(name).asText(), "text/xml");
 
-                            coordCarto = new CoordCarto().setFromDegreeGeo(longitude,latitude, altitude);
+                    var longitude = Number(doc.getElementsByTagName("longitude")[0].childNodes[0].nodeValue);
+                    var latitude = Number(doc.getElementsByTagName("latitude")[0].childNodes[0].nodeValue);
+                    var altitude = Number(doc.getElementsByTagName("altitude")[0].childNodes[0].nodeValue);
 
-                        }
-                    }
+                    coordCarto = new CoordCarto().setFromDegreeGeo(longitude,latitude, altitude);
+                }
+            }
 
-                    collada.coorCarto = coordCarto;
-
-                    resolve(collada);
-
-                };
-
-                xhr.onerror = function() {
-
-                    reject(Error("Error KMZLoader"));
-
-                };
-
-                xhr.send(null);
+            collada.coorCarto = coordCarto;
+            return collada;
+        }
 
+        KMZLoader.prototype.load = function(url) {
+            return fetch(url).then(function(response) {
+                return this.parseCollada(response.arrayBuffer(), this.colladaLoader);
```

Comment:
`response.arrayBuffer()` returns a `Promise`, so it should probably be "cascaded" as in `IoDriver_XBil`.

Not sure we need to pass the `ColladaLoader` as argument (AFAICT, `scopeLoader` was used much like `var that = this;` as the callbacks didn't use `bind(this)`)